src/ripple/conditions/impl/condition.cpp

```diff
@@ -0,0 +1,107 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/basics/contract.h>
+#include <ripple/conditions/condition.h>
+#include <ripple/conditions/fulfillment.h>
+#include <ripple/conditions/hashlock.h>
+#include <ripple/conditions/impl/utils.h>
+#include <boost/regex.hpp>
+#include <boost/optional.hpp>
+#include <vector>
+
+namespace ripple {
+namespace cryptoconditions {
+
+boost::optional<condition_t>
+load_condition(std::string const& s)
+{
+    // CHECKME: custom parser maybe? probably faster but
+    // more work and probability of error.
+
+    // TODO: use two regex: one that accepts anything the
+    // standard supports and one which accepts only what
+    // we support. Parse with both for improved errors?
+    static boost::regex const re_current (
+            "^"                            // start of line
+            "cc\\:"                        // 'cc' for cryptocondition
+            "([1-9a-f][0-9a-f]{0,3}|0):"   // type (hexadecimal)
+            "([1-9a-f][0-9a-f]{0,15}):"    // feature bitmask (hexadecimal)
```

Comment:
You might be able to use the `[:xdigit:]` character class here (would have to check about case sensitivity and which `std::regex` grammars supports it.